src/ray/gcs/gcs_client/usage_stats_client.cc

```diff
@@ -0,0 +1,44 @@
+// Copyright 2022 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/gcs/gcs_client/usage_stats_client.h"
+
+namespace ray {
+namespace gcs {
+UsageStatsClient::UsageStatsClient(const std::string &gcs_address,
+                                   instrumented_io_context &io_service)
+    : io_service_(io_service) {
+  GcsClientOptions options(gcs_address);
+  gcs_client_ = std::make_unique<GcsClient>(options);
+}
+
+void UsageStatsClient::RecordExtraUsageTag(const std::string &key,
+                                           const std::string &value) {
+  if (!gcs_client_->IsConnected()) {
```

Comment:
Why don't we just assert? I think this should never be true if things are used correctly

Reply:
hmm actually nvm. I just realized this uses a different instance